replica/distributed_loader.cc

```diff
@@ -160,10 +160,28 @@ collect_all_shared_sstables(sharded<sstables::sstable_directory>& dir) {
     auto coordinator = this_shard_id();
     // We will first move all of the foreign open info to temporary storage so that we can sort
     // them. We want to distribute bigger sstables first.
-    co_await dir.invoke_on_all([&info_vec, coordinator] (sstables::sstable_directory& d) -> future<> {
+    co_await dir.invoke_on_all([&info_vec, coordinator, owned_ranges_ptr] (sstables::sstable_directory& d) -> future<> {
+        auto shared_sstables = d.retrieve_shared_sstables();
+        sstables::sstable_directory::sstable_info_vector need_cleanup;
+        if (owned_ranges_ptr) {
+            const auto& owned_ranges = *owned_ranges_ptr;
+            co_await d.do_for_each_sstable([&] (sstables::shared_sstable sst) -> future<> {
+                if (needs_cleanup(sst, owned_ranges, sst->get_schema())) {
+                    need_cleanup.push_back(co_await sst->get_open_info());
+                }
```

Comment:
needs_cleanup() can have a high false positive rate, but we'll only include cleanup sstables for reshard on refresh, which is good. document it?!

Reply:
I'm not sure I understand your comment.
Where do you want it documented?

Reply:
I am saying that needs_cleanup() may falsely state that a sstable needs cleanup. If we were able to run cleanup + resharding on boot (I know we cannot), it could happen we'd trigger reshard unnecessarily. But this is resharding on refresh (sstable loading via upload dir), which will usually require resharding anyway. Documenting it somewhere could be good.

Reply:
Added comment in next version (v3)